include/LightGBM/dataset.h

```diff
@@ -337,6 +337,12 @@ class Dataset {
         const int group = feature2group_[feature_idx];
         const int sub_feature = feature2subfeature_[feature_idx];
         feature_groups_[group]->PushData(tid, sub_feature, row_idx, feature_values[i]);
+        if (has_raw_) {
+          int feat_ind = numeric_feature_map_[feature_idx];
+          if (feat_ind >= 0) {
+            raw_data_[feat_ind][row_idx] = feature_values[i];
```

Comment:
why not use row-first raw data? I guess linear model is faster with row-first data?

Reply:
Yes, see previous comment: https://github.com/microsoft/LightGBM/pull/3299#discussion_r482742127

Reply:
See previous comment: https://github.com/microsoft/LightGBM/pull/3299#discussion_r482742127

Reply:
great, i missed it.